CRM/Admin/Page/Extensions.php

```diff
@@ -270,7 +270,7 @@ public function formatRemoteExtensionRows($localExtensionRows) {
       );
       if (isset($localExtensionRows[$info->key])) {
         if (array_key_exists('version', $localExtensionRows[$info->key])) {
-          if (version_compare($localExtensionRows[$info->key]['version'], $info->version, '<')) {
+          if (version_compare($localExtensionRows[$info->key]['version'] ?? '', $info->version, '<')) {
```

Comment:
The two lines before this seem intended to at least partly guard against this already. Maybe just update them to a single `if (!empty($localExtensionRows[$info->key]['version']))`? (version wouldn't be `0`)

Reply:
Sorry my reply is getting way to long! But hopefully it makes things clearer and easier to engage with.

Short Answer: 
Yes agree that would be more concise but that would change the behaviour. I was hoping to just squash the notice instead of figure out the wider implications :) . But happy to engage with wider implications if needed.

Long Answer:

Currently the check for an upgraded version runs if:
` $localExtensionRows[$info->key]` exists and has an array key of version.

In the case of missing code:
`$localExtensionRows[$info->key]['version']  == null`

I think this is because createExtendedInfo ends up calling fillMissingKeys which populates any missing keys with null values then passes them into this function. However this function is public and no expectation is give for the structure of localExtensionRows... (perhaps this is the issue!)


Anyway in the above where the value is null the if's pass and we run the check. 

If we use 

`(!empty($localExtensionRows[$info->key]['version']))`

this would mean that the if would prevent the version check from running, where as before it would run.

Reply:
Ok good point.
I see it still has the popup about a missing extension so you still get some notice that something's wrong.